### PR TITLE
[clickhouse] upload logs on test failure

### PR DESCRIPTION
**Pull request by sunshowers**
Part of trying to figure out #4779 -- we now create ClickHouse logs in a
well-named directory which gets not-cleaned up (and therefore uploaded) at the end.

Tested this by introducing a failure in a test, and then seeing the following files in `/tmp`:

```
% find test_all-c4d6fc05c1fc48de-test_local_users.3329283.1-clickhouse-I4Af08 
test_all-c4d6fc05c1fc48de-test_local_users.3329283.1-clickhouse-I4Af08
test_all-c4d6fc05c1fc48de-test_local_users.3329283.1-clickhouse-I4Af08/log
test_all-c4d6fc05c1fc48de-test_local_users.3329283.1-clickhouse-I4Af08/clickhouse-server.log
test_all-c4d6fc05c1fc48de-test_local_users.3329283.1-clickhouse-I4Af08/clickhouse-server.errlog
```

Depends on https://github.com/oxidecomputer/dropshot/pull/878.

### CODE REVIEW

**Comment by sunshowers** [comment]
cc @bnaecker since you were discussing uploading ClickHouse logs

**Comment by sunshowers** [comment]
I've been trying to repro the failures on this PR but sadly none of the runs have repro'd it so far. Hoping that landing this on main works.

File: test-utils/src/dev/clickhouse.rs [hunk 1]
```diff
@@ -294,10 +301,14 @@ struct ClickHouseDataDir {
 }
 
 impl ClickHouseDataDir {
-    fn new() -> Result<Self, anyhow::Error> {
-        // Keepers do not allow a dot in the beginning of the directory, so we must
-        // use a prefix.
-        let dir = Utf8TempDir::with_prefix("clickhouse-")
+    fn new(logctx: &LogContext) -> Result<Self, anyhow::Error> {
+        // TODO: replace with log_prefix_for_test once
+        // https://github.com/oxidecomputer/dropshot/pull/878 is merged.
```
Comment (smklein) [inline test-utils/src/dev/clickhouse.rs:306]: This is done, right?

**Review by smklein — APPROVED** [review]

File: test-utils/src/dev/clickhouse.rs (hunk 1 above)
Comment (sunshowers) [reply test-utils/src/dev/clickhouse.rs:306]: Yeah, thought I removed this comment. Huh

**Pull request merged**